sql/parser/builtins.go

```diff
@@ -1145,46 +1146,63 @@ func regexpReplace(ctx EvalContext, s, pattern, to, sqlFlags string) (Datum, err
 	return DString(newString.String()), nil
 }
 
+var flagToByte = map[syntax.Flags]byte{
+	syntax.FoldCase: 'i',
+	syntax.DotNL:    's',
+}
+
+var flagToNotByte = map[syntax.Flags]byte{
+	syntax.OneLine: 'm',
+}
+
 // regexpEvalFlags evaluates the provided Postgres regexp flags in
 // accordance with their definitions provided at
 // http://www.postgresql.org/docs/9.0/static/functions-matching.html#POSIX-EMBEDDED-OPTIONS-TABLE.
 // It then returns an adjusted regexp pattern.
 func regexpEvalFlags(pattern, sqlFlags string) (string, error) {
-	goReFlags := map[rune]struct{}{'s': {}, 'm': {}}
+	flags := syntax.DotNL
 
-	for _, flag := range sqlFlags {
-		switch flag {
+	for _, sqlFlag := range sqlFlags {
+		switch sqlFlag {
 		case 'g':
-			// Ignore for now, valid flag that should be handled elsewhere.
+			// Handled in `regexpReplace`.
 		case 'i':
-			goReFlags['i'] = struct{}{}
+			flags |= syntax.FoldCase
 		case 'c':
-			delete(goReFlags, 'i')
+			flags &= ^syntax.FoldCase
```

Comment:
`&= ^` is more easier for me to understand though, so I dont have a strong opinion either way.